Assignment4/santi-gv-190031/task06.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/FacultadInformatica-LinkedData/Curso2022-2023/blob/master/Assignment4/course_materials/notebooks/Task06.ipynb
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+g.add((ns.Reseacher,RDFS.subClassOf, ns.Person))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+janeUri = ns.JaneSmith
+g.add((janeUri, RDF.type, ns.Researcher))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the fullName, given and family names**"""
+
+# TO DO
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+fullName = Literal("Jane Smith")
+given = Literal("Jane")
+family = Literal("Smith")
+g.add((janeUri, VCARD.FN, fullName))
+g.add((janeUri, VCARD.Given, given))
+g.add((janeUri, VCARD.Family, family))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+johnUri = ns.JohnSmith
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((johnUri, VCARD.work, ns.UPM))
```

Comment:
vcard does not define "work" as a property. Define your own

Reply:
resolved